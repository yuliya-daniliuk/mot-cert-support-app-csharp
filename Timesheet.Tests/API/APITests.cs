using ApprovalTests;
using static RestAssured.Dsl;
using Timesheet.Models.Auth;

namespace Timesheet.Tests.API;

public class APITests
{
    [Test]
   public void TestGettingProject2()
   {
        Credentials credentials = (Credentials)Given()
                                       .Body("{\"email\":\"admin@test.com\",\"password\":\"password123\"}")
                                       .ContentType("application/json")
                                       .Post("http://localhost:8080/v1/auth/login")
                                       .DeserializeTo(typeof (Credentials));
        string token = credentials.Token;
        
        HttpResponseMessage response = Given()
                                       .Header("Authorization", "Bearer " + token)
                                       .Get("http://localhost:8080/v1/project/2")
                                       .Then()
                                       .Extract()
                                       .Response();

        Approvals.Verify(response.Content.ReadAsStringAsync().Result);
   }

}
