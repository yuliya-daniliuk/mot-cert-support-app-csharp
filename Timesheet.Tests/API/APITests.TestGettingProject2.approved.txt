{"name":"Project 2","description":"This is the second project that I want created","total":0,"entries":[]}